docs_sphinx/user/numerical_integration.rst

```diff
@@ -60,3 +74,80 @@ objects will use ``'linear'``, ``'exponential_euler'``, ``'rk2'`` or ``'heun'``.
+  If the resulted estimated error exceeds the set error bounds, the simulation
+  is aborted. When using cython or weave this is reported with an `IntegrationError`.
+  Defaults to ``True``.
+* ``'absolute_error'``: each of the methods has a way of estimating the error that
+  is the result of using numerical integration. You can specify the maximum size of this
+  error to be allowed for any of the to-be-integrated variables in base units with this
+  keyword. Note that giving very small values makes the simulation slow and might result
+  in unsuccessful integration. In the case of using the ``'absolute_error_per_variable'``
+  option, this is the error for variables that were not specified individually.
+  Defaults to 1e-6.
+* ``'absolute_error_per_variable'``: specify the absolute error per variable in its own
+  units. Variables for which the error is not specified use the error set with the
+  ``'absolute_error'`` option.
+  Defaults to None.
+* ``'max_steps'``: The maximal number of steps that the integrator will take within a
+  single "Brian timestep" in order to reach the given error criterion. Can be set to
+  0 to not set any limits. Note that without limits, it can take a very long time
+  until the integrator figures out that it cannot reach the desired error level. This
+  will manifest as a simulation that appears to be stuck.
+  Defaults to 100.
+* ``'use_last_timestep'``: with the ``'adaptable_timestep'`` option set to True, GSL tries
+  different time steps to find a solution that satisfies the set error bounds.
+  It is likely that for Brian's next time step the GSL time step
+  will be somewhat similar per neuron (e.g. active neurons will have a shorter GSL time step
+  than inactive neurons). With this option set to True, the time step GSL found to satisfy
+  the set error bounds is saved per neuron and given to GSL again in Brian's next time step.
+  This also means that the final time steps are saved in Brian's memory and can thus
+  be recorded with the `StateMonitor`: it can be accessed under ``'_last_timestep'``.
+  Note that some extra memory is required to keep track of the last time steps.
+  Defauls to True.
+* ``'save_failed_steps'``: if ``'adaptable_timestep'`` is set to True,
+  each time GSL tries a time step and it results in an estimated
+  error that exceeds the set bounds, one is added to the ``'_failed_steps'`` variable. For
+  purposes of investigating what happens within GSL during an integration step, we offer
+  the optionality of saving this variable.
+  Defaults to False.
+* ``'save_step_count'``: the same goes for the total number of GSL steps taken in a single
+  Brian time step: this is optionally saved in the ``'_step_count'`` variable.
+  Defaults to False.
+
+Note that at the moment recording ``'_last_timestep'``, ``'_failed_steps'``, or ``'_step_count'``
+requires a call to `run` (e.g. with 0 ms) to trigger the code generation process, before the
+call to `StateMonitor`.
```

Comment:
That would be quite tricky... The problem here is that we add something to the `variables` of a `NeuronGroup`, but during the code generation process of the state updater -- this is something that we can hopefully fix with a refactoring of the state update/code generation process.
In general, I think it could be good to modify `StateMonitor` so that it is initialized in `before_run`, similar to what we do for the `SpikeQueue`. This might be a way to deal with the `record=True` issue for synapses on standalone (#528). On the other hand, this comes at the cost of making the code harder to follow/extend.
Either way, even this would not be enough here. We'd also make sure that `StateUpdater.before_run` is executed before `StateMonitor.before_run`. At the moment, we don't have any kind of ordering of the `before_run` operations. It might be good to have something like this in the long run, but it's definitely not an easy fix...